packages/blade/src/components/Box/index.ts

```diff
@@ -0,0 +1 @@
+export { default } from './Box';
```

Comment:
Maybe let's use named exports.
It's 10x easier to reason about named import and avoid bugs and confusion. 

```
export * from './Box';
```

Reply:
In consumer side we are anyway doing named export. That's no the issue. 

```
export { default as Box } from './Box' 
```

The main thing is the internal blade components. 
- With default exports it's hard to reason about exports
- If we rename a component we won't get any error that we renamed and might continue using the older name


```
// Box.d.ts
export { default as Box } from './Box.d';
export * from './Box';
```

vs

```
// Box.d.ts
export * from './Box';
```

Reply:
>With default exports it's hard to reason about exports

Not sure what you mean here

>If we rename a component we won't get any error that we renamed and might continue using the older name

Yes, that's a blessing and a curse 😅 


Take a look at BaseText or Icons

We do
```
import BaseText from '../../BaseText';
import CreditCardIcon from '../../CreditCardIcon';
```

Honestly, I don't mind named exports - just as long as we do the same things everywhere. 

If we have a strong enough reason to move to named exports, lets do that everywhere. But don't want to do a default export in some places and named in others. Lets take this up separately on whether or not we want to move everything to named exports? That would require change in multiple places and should be handled in another PR

Reply:
Yeah we can decide upon that separately and should also add eslint no default exports once we go full in with named exports.

Reply:
Yeah, lets have a discussion around this. Would like to understand better why we'd prefer named exports for Box, BaseText, Icons, etc over default

Reply:
I would personally prefer using default exports for this purpose 😄